packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -423,49 +555,132 @@ const QueryEditorHeaderContent = observer(
         .catch(applicationStore.alertUnhandledError);
     };
 
+    const toggleAssistant = (): void =>
+      applicationStore.assistantService.toggleAssistant();
+
+    const queryDocEntry = applicationStore.documentationService.getDocEntry(
+      LEGEND_APPLICATION_DOCUMENTATION_KEY.TUTORIAL_QUERY_BUILDER,
+    );
+
+    const openQueryTutorial = (): void => {
+      if (queryDocEntry?.url) {
+        applicationStore.navigationService.navigator.visitAddress(
+          queryDocEntry.url,
+        );
+      }
+    };
+
+    const extraHelpMenuContentItems = applicationStore.pluginManager
+      .getApplicationPlugins()
+      .flatMap((plugin) => plugin.getExtraHelpMenuItemsRender?.() ?? [])
+      .map((item) => <>{HelpMenuContentItemRenderer(item)}</>);
+
     return (
       <div className="query-editor__header__content">
-        <div className="query-editor__header__content__main" />
+        <div
+          className="query-editor__header__content__main query-editor__header__content__title"
+          title="Query title"
+        >
+          {editorStore.titleOfQuery}
+        </div>
         <div className="query-editor__header__actions">
-          {editorStore instanceof ExistingQueryEditorStore &&
-            applicationStore.pluginManager
-              .getApplicationPlugins()
-              .flatMap(
-                (plugin) =>
-                  plugin.getExtraExistingQueryActionRendererConfiguration?.() ??
-                  [],
-              )
-              .map((actionConfig) => (
-                <Fragment key={actionConfig.key}>
-                  {actionConfig.renderer(editorStore, queryBuilderState)}
-                </Fragment>
-              ))}
-          <button
+          {editorStore instanceof ExistingQueryEditorStore
+            ? applicationStore.pluginManager
+                .getApplicationPlugins()
+                .flatMap(
+                  (plugin) =>
+                    plugin.getExtraExistingQueryActionRendererConfiguration?.() ??
+                    [],
+                )
+                .map((actionConfig) => (
+                  <Fragment key={actionConfig.key}>
+                    {actionConfig.renderer(editorStore, queryBuilderState)}
+                  </Fragment>
+                ))
+            : applicationStore.pluginManager
+                .getApplicationPlugins()
+                .flatMap(
+                  (plugin) =>
+                    plugin.getExtraNonExistingQueryActionRendererConfiguration?.() ??
+                    [],
```

Comment:
Instead of having `getExtraNonExistingQueryActionRendererConfiguration` and `getExtraExistingQueryActionRendererConfiguration` you can just have `getExtraQueryActionRendererConfiguration` and pass `editorStore` to this and let the code in the extension to choose what it wants to render. If the editorStore is `ExistingQueryEditorStore` render the logic specific to it and if it is not `ExistingQueryeditorStore` render the logic for non-existing queries